### ISSUE TITLE
Discussion: Flexibility of the new bank0 and bank1 VRAM banks

### ISSUE THREAD

**Issue by joshgoebel** [nesbox/TIC-80#1716]
I don't think you considered it in your design, but it's been mentioned elsewhere... that one could imagine a game where you have a single screen, but that you might want to render to bank0 or bank1 and then *selectively* choose which is shown to the player.  IE, no transparency - two entirely different screens that you want to CHOOSE from each tic.

Right now it seems you could HALF do this with:

```lua
-- render entirely from bank 1
vbank(1)
poke(0x03FF8, 0xFF) -- or any other invalid palette entry

-- render entirely from bank 0
vbank(1)
poke(0x03FF8, 0) 
cls(0)
```

So one can switch from bank0 to 1 simply by setting an invalid transparent color (ie, NO transparency)... but there is no good way to render bank 0 *without* rendering bank 1... could this use case be considered?  Could we use a further overload of the new `clear` register to do this... Perhaps just a fixed set of magic values:

```
0xFF - no transparency, bank 1 is rendered
0xFE - bank 1 overlay is disabled, only bank 0 is rendered
```

**Comment by joshgoebel** [comment]
Great job on getting rid of 99% of the magic around OVR... but this "bank 1 is ALWAYS rendered on top of bank 0, period" smells like a bit of a hold over over from the OVR days - rather than a strict requirement... seems it would be much nicer if we let the programmer decide how bank 0 and bank 1 were rendered rather than hard-wiring it so they can only be used for a single purpose.

A 16-bit palette bitmask (vs 4 bit static value) is another idea... so you could set it to `0xFFFF` to mask ALL colors as transparent (hence rendering only bank 0 and leaving bank 1 entirely "hidden").

**Comment by nesbox** [comment]
We can allow users to define which layer will be rendered first and second. So, the currently selected vbank will be the first layer and so on...

**Comment by joshgoebel** [comment]
> So, the currently selected vbank will be the first layer and so on...

Ok, now that you've changed this someone could write an "every other frame, every other bank" simply by always letting the "overlay" be drawn 100% opaque [with invalid transparency]... thus the layer you'd want active at any given moment would actually be the "hidden" layer... weird, but easy enough to grasp I think once it's explained.

So people playing with vbanks just have to be careful?  And our advice is to just use them consistently?

**Comment by nesbox** [comment]
Yes, people have to be careful, the best is if users switch the bank, draw in it, and immediately switch back.
